src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -1130,113 +1153,223 @@ export function makeHeatmapTracksMobxPromise(
+                            oncoprint.removeHeatmapTracksByMolecularProfileId(
+                                molecularProfileId
+                            );
+                        } else {
+                            oncoprint.setHeatmapTracks(
+                                molecularProfileId,
+                                newEntities
+                            );
+                        }
+                    }
+
+                    if (
+                        trackGroup === undefined &&
+                        oncoprint.sortMode.type === 'heatmap' &&
+                        oncoprint.sortMode.clusteredHeatmapProfile ===
+                            molecularProfileId
+                    ) {
+                        oncoprint.sortByData();
+                    }
+                });
+
+                let dataForTrack: {
+                    uniqueSampleKey: string;
+                    uniquePatientKey: string;
+                    value: number;
+                }[] = [];
+
+                if (profileType === AlterationTypeConstants.MUTATION_EXTENDED) {
+                    const mutationPromise = oncoprint.props.store.annotatedMutationCache.get(
+                        {
+                            entrezGeneId: query.entrezGeneId,
+                        }
+                    );
+
+                    const mutations = mutationPromise.result;
+
+                    if (mutations) {
+                        const profileMutations = mutations.filter(
+                            m => m.molecularProfileId === molecularProfileId
+                        );
+
+                        dataForTrack = profileMutations
+                            .map(m => {
+                                let vafValue: number | null = null;
+                                const vafReport = getVariantAlleleFrequency(m);
+
+                                if (
+                                    vafReport &&
+                                    typeof vafReport.vaf === 'number' &&
+                                    !isNaN(vafReport.vaf)
+                                ) {
+                                    vafValue = vafReport.vaf;
+                                } else {
+                                    const { tumorAltCount, tumorRefCount } = m;
+                                    if (
+                                        typeof tumorAltCount === 'number' &&
+                                        typeof tumorRefCount === 'number'
+                                    ) {
+                                        const totalCount =
+                                            tumorAltCount + tumorRefCount;
+                                        if (totalCount > 0) {
+                                            vafValue =
+                                                tumorAltCount / totalCount;
+                                        }
+                                    }
```

Comment:
I am surprised that we need this logic. I would assume that it's taken care of by the `getVariantAlleleFrequency` function.